### PR TITLE
Allow setting fill colour for bar chart data points

### LINKED ISSUE #1400: Bar chart fillcolor not working
This is:

```
- [* ] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
If I set the colors for a BarChar, I'm expecting the lines to have those colors.

### What is the current behavior?
BarChar don't have those colors

### What are the steps to reproduce?
Take the sample file 33_Chart_create_bar_stacked.php, and add a color.

just change this line 
 $dataSeriesValues = array(
                    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, 'Worksheet!$D$2:$D$5', NULL, 4, array(), null, '000000')) ;

### Which versions of PhpSpreadsheet and PHP are affected?
 the latest

### PR DESCRIPTION
**Pull request by DanWin**
This is:

- [X] a bugfix
- [ ] a new feature

Checklist:

- [ ] Changes are covered by unit tests
- [X] Code style is respected
- [X] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [X] CHANGELOG.md contains a short summary of the change
- [ ] Documentation is updated as necessary

### Why this change is needed?

This will fix #1400 #1433 and #2183

### CODE REVIEW

File: src/PhpSpreadsheet/Writer/Xlsx/Chart.php [hunk 1]
```diff
@@ -1082,7 +1082,7 @@ private function writePlotGroup(?DataSeries $plotGroup, $groupType, XMLWriter $o
             //    Values
             $plotSeriesValues = $plotGroup->getPlotValuesByIndex($plotSeriesRef);
 
-            if (($groupType == DataSeries::TYPE_PIECHART) || ($groupType == DataSeries::TYPE_PIECHART_3D) || ($groupType == DataSeries::TYPE_DONUTCHART)) {
+            if (in_array($groupType, [DataSeries::TYPE_PIECHART, DataSeries::TYPE_PIECHART_3D, DataSeries::TYPE_DONUTCHART, DataSeries::TYPE_BARCHART, DataSeries::TYPE_BARCHART_3D])) {
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Writer/Xlsx/Chart.php:1085]: `in_array()` should always be used in strict mode

**Comment by MarkBaker** [comment]
Is it possible to provide some unit tests to verify this for all chart types that are affected?

**Review by MarkBaker — CHANGES_REQUESTED** [review]

**Review by MarkBaker — CHANGES_REQUESTED** [review]
Please address requested change; and unit testing

**Comment by DanWin** [comment]
Agreed, `in_array()` should be in strict mode. I saw it mostly being non-strict in other parts of the code and the previous checks were also non-strict and thus kept it the same way.
As for unit testing, I'll need help with that. I've never written those and I'm not sure how to write tests for this case.

**Comment by MarkBaker** [comment]
> Agreed, `in_array()` should be in strict mode. I saw it mostly being non-strict in other parts of the code and the previous checks were also non-strict and thus kept it the same way.

Thanks. A lot of the code that uses `in_array()` was originally written for earlier versions of PHP, before the strict mode flag was added. Now, we always try to apply it when we're working on those parts of the codebase where it isn't yet applied (scouting rules).

> As for unit testing, I'll need help with that. I've never written those and I'm not sure how to write tests for this case.

There's some examples on how to approach testing in `/tests/PhpSpreadsheetTests/Writer/Xlsx/*` where we use the writer calls to return the XML to the test rather than actually saving to a file; and can then assert that the XML contains the markup that's expected.
e.g. `ConditionalTest.php` and the `testWriteSimpleCellConditionalFromWizard()` method.

Though you might want to create a `Charts` subfolder under `.../Writer/Xlsx/`, and put the test in there.
 
Something like:
```php
namespace PhpOffice\PhpSpreadsheetTests\Writer\Xlsx\Charts;

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheetTests\Functional\AbstractFunctional;

class BarChartTest extends AbstractFunctional
{
    public function testWriteBarChartWithUserDefinedFillColors(): void
    {
        $spreadsheet = new Spreadsheet();
        $worksheet = $spreadsheet->getActiveSheet();

        // Code to create a bar chart with user-defined fill colours goes here

        $writer = new Xlsx($spreadsheet);
        $writerWorksheet = new Xlsx\Worksheet($writer);
        $data = $writerWorksheet->writeWorksheet($worksheet, []);

        $expected = <<<XML
//xml for a bar chart with user-defined fill colours goes here
XML;
        self::assertStringContainsString($expected, $data);
    }
}
```

**Comment by oleibman** [comment]
This PR will be superseded by #2906, including tests. Converting this to draft; and will close it if 2906 is merged.

**Comment by oleibman** [comment]
Superseded by 2906, which was just merged.

**Pull request closed without merging**